### PR TITLE
Support websocket connections over HTTP proxy

### PR DESCRIPTION
**Pull request by l-n-s**

### CODE REVIEW

**Comment by l-n-s** [comment]
Original ticket: https://trac.i2p2.de/ticket/2422

File: libi2pd_client/HTTPProxy.cpp [hunk 1]
```diff
@@ -219,7 +219,9 @@ namespace proxy {
 		/* replace headers */
 		req.UpdateHeader("User-Agent", "MYOB/6.66 (AN/ON)");
 		/* add headers */
-		req.UpdateHeader("Connection", "close"); /* keep-alive conns not supported yet */
+		auto h = req.GetHeader ("Connection");
+		if (h.find("upgrade") == std::string::npos && h.find("Upgrade") == std::string::npos)
```
Comment (orignal) [inline libi2pd_client/HTTPProxy.cpp]: Better to try to find "pgrade" and then compare lowercase of [-1] with 'u'

Comment (l-n-s) [reply libi2pd_client/HTTPProxy.cpp]: Something like that?

`if (std::tolower(h[x - 1], std::locale("C")) == 117) do_stuff();`

Comment (l-n-s) [reply libi2pd_client/HTTPProxy.cpp]: check now

**Pull request merged**